src/main/java/org/eclipse/yasson/internal/model/PropertyValuePropagation.java

```diff
@@ -116,12 +116,20 @@ private boolean isFieldVisible(Field field, Method method) {
         }
         Boolean accessible = isVisible(strategy -> strategy.isVisible(field), field, method);
         //overridden by strategy, or anonymous class (readable by spec)
-        if (accessible && (!Modifier.isPublic(field.getModifiers()) || field.getDeclaringClass().isAnonymousClass())) {
+        if (accessible && (
+                !Modifier.isPublic(field.getModifiers())
+                        || field.getDeclaringClass().isAnonymousClass()
+                        || isClassPackagePrivate(field.getDeclaringClass()))) {
             overrideAccessible(field);
         }
         return accessible;
     }
 
+    private boolean isClassPackagePrivate(Class<?> declaringClass) {
+        int modifiers = declaringClass.getModifiers();
+        return !(Modifier.isPublic(modifiers) || Modifier.isProtected(modifiers) || Modifier.isPrivate(modifiers));
```

Comment:
do we also need to restrict private? If package protected does not work, I would expect private to fail in the same way.

Reply:
From my point of view it is proper way to cover all possible values simply just to be sure. :-)

Reply:
but if the class is private, we still need to override access, and probably protected too. So I'm thinking this line should just be:
`return !Modifier.isPublic(modifiers)`
which would eliminate the need for the helper method since it would not be much code now

Reply:
Oh I see your point now. If I am getting this right according to spec we can allow only protected and public in case of nested (or static nested) classes. In case non-nested class it is up to us I guess. I will make a slight change to this code.